CMakeLists.txt

```diff
@@ -77,7 +77,15 @@ OPTION(BUILD_TESTS "xtensor-io test suite" OFF)
 OPTION(DOWNLOAD_GTEST "build gtest from downloaded sources" OFF)
 OPTION(DOWNLOAD_GBENCHMARK "download google benchmark and build from source" ON)
 
-# Find OpenImageIO
+OPTION(HAVE_OIIO "require OpenImageIO dependency for image file support" OFF)
+OPTION(HAVE_SndFile "require SndFile for audio file support" OFF)
+OPTION(HAVE_ZLIB "require ZLIB for npz file support" OFF)
+OPTION(HAVE_HighFive "require HighFive for HDF5 file support" OFF)
+OPTION(HAVE_Blosc "require Blosc for Blosc file support" OFF)
+OPTION(HAVE_GDAL "require GDAL for geospatial raster file support" OFF)
+OPTION(HAVE_storage_client "require storage_client for Google Cloud Storage IO handler support" OFF)
+OPTION(HAVE_AWSSDK "require AWSSDK for AWS S3 IO handler support" OFF)
+
```

Comment:
This makes building the tests a bit cumbersome. We could add a "-DHAVE_ALL_DEPS" option to enable them all,.

A more flexible idea would be to pass the list of optional dependencies to the cmake command:

```bash
cmake -DCMAKE_PREFIX_PATH=$CONDA_PREFIX -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX -DDOWNLOAD_GTEST=ON -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX "-DOPTIONAL_DEPENDENCIES=OIIO;SndFile;zlib;HighFive;Blosc;GDAL;storage_client;AWSSDK" $SRC_DIRECTORY
```

Notice that this dependency list can live with the DHAVE_ALL_DEPS option.